parser-typechecker/src/Unison/Codebase/FileCodebase/CopyFilterIndex.hs

```diff
@@ -84,22 +84,9 @@ syncToDirectory' :: forall m
   => CodebasePath
   -> CodebasePath
   -> Branch m
-  -> m (Branch m)
-syncToDirectory' srcPath destPath branch =
+  -> m ()
+syncToDirectory' srcPath destPath newRemoteRoot@(Branch c) =
   flip State.evalStateT mempty $ do
-    newRemoteRoot@(Branch c) <- lift $
-      ifM (codebaseExists destPath)
-        (getRootBranch destPath >>= \case
-          Right existingDestRoot -> Branch.merge branch existingDestRoot
-          -- The destination codebase doesn't advertise a root branch,
-          -- so we'll just use ours.
-          Left Codebase.NoRootBranch -> pure branch
-          Left (Codebase.CouldntLoadRootBranch h) -> fail $
-            "I was trying to merge with the existing root branch at " ++
-            branchPath destPath h ++ ", but the file was missing."
-          )
-        -- else there was no existing codebase structure at `destPath` so whatev
-        (pure branch)
```

Comment:
Instead of trying to do this error handling and merging during the sync (which we only wanted for `pull` anyway, and not for `push`), we do it before this call. 

The load error handling now happens in `Git.viewRemoteBranch'`, a "fast-forward only" / `Branch.before` check happens in `Git.pushGitRootBranch`, and any merging has to happen in `HandleInput.hs`, not here.